inbm-lib/inbm_common_lib/utility.py

```diff
@@ -233,8 +233,8 @@ def validate_file_type(path: List[str]) -> None:
     """
     logger.debug("Supported file type prefixes: {0}".format(VALID_MAGIC_FILE_TYPE_PREFIXES))
 
-    tarball_list: List[str] = [x for x in path if (not str(x).endswith('.mender') and tarfile.is_tarfile(x))]
-    extracted_file_list: List[str] = []
+    tarball_list: list[str] = [x for x in path if (not str(x).endswith('.mender') and tarfile.is_tarfile(x))]
+    extracted_file_list: list[str] = []
 
     for tarball in tarball_list:
         with tarfile.open(tarball) as tar:
```

Comment:
**<h2>$\color{#FFA500}Improvements: Code Robustness$</h2>**
**Suggestion:** The validate_file_type function should use a context manager when opening the tarball to ensure it is always closed properly, even if an exception occurs.
**NOTE:** Ensure that the suggested code is comprehensive. This serves as an example of the code you could use. The user should verify it after accepting the suggestion.
```suggestion
    for tarball in tarball_list:
        with tarfile.open(tarball) as tar:
            logger.debug(f"Extract {tarball}.")
            safe_extract(tar, path=TEMP_EXT_FOLDER)
            extracted_files = tar.getmembers()
```